### ISSUE TITLE
`CacheDirective` enum is missing some directives

### ISSUE THREAD

**Issue by ssokolow** [actix/actix-web#2666]
## Expected Behavior
I should be able to use `CacheDirective::Immutable` for implementing `Cache-Control` headers for the cache-busting pattern.

## Current Behavior
I have to use `CacheDirective::Extension("immutable".to_owned(), None)`

## Possible Solution
Add `CacheDirective` variants for the following variants [listed on MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) but missing from the enum:

* must-understand
* immutable
* stale-while-revalidate
* stale-if-error

## Your Environment

- Rust Version (I.e, output of `rustc -V`): rustc 1.59.0 (9d1b2106e 2022-02-23)
- Actix Web Version: 4.0.1

**Comment by robjtede** [comment]
shame this wasn't caught before stable v4 :/

you _can_ implement your own typed headers though

**Comment by ssokolow** [comment]
> shame this wasn't caught before stable v4 :/

Yeah, especially since it was just coincidence that I happened to be preparing to get back to work on the project in question around the v4 release.

To be honest, I'm surprised you didn't mark that enum nonexhaustive.

> you *can* implement your own typed headers though

Not really worth the trouble. I only use it in one place (a helper macro for asset routes), so it's more about moving the responsibility for ensuring typos don't creep in to somewhere with more eyeballs.

**Comment by robjtede** [comment]
Since this can't be fixed until v5, I added an updated verison to `actix-web-lab`: <https://docs.rs/actix-web-lab/0.15.0/actix_web_lab/header/struct.CacheControl.html>.

**Comment by AverageHelper** [comment]
Should `CacheDirective` (and other exported enums) be marked `non_exhaustive` in a possible v5? This way at least, if a variant is forgotten about or the spec is extended by a future RFC, adding support isn't as breaking of a change from then on..

EDIT: ah, I see it is so in the `actix-web-lab` version. Neat!
